jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l

```diff
@@ -80,28 +88,39 @@
     (return-from undock nil))
   (send (send *undock-action* :get-result) :undocked))
 
-(defun go-to-spot (name &optional (relative-coords (make-coords)))
+(defun go-to-spot (name &optional (relative-coords (make-coords)) &key (undock-rotate nil))
   ;; undock if fetch is docking
   (unless (boundp '*ri*)
     (require :fetch-interface "package://fetcheus/fetch-interface.l")
     (fetch-init))
-  (dotimes (i 3)
-    (let ((msg (one-shot-subscribe "/battery_state" power_msgs::batterystate :timeout 1500)))
-      ;; You may fail to subscribe /battery_state
-      ;; because of message md5 difference between melodic and indigo.
-      (unless msg
-        (ros::ros-warn "Failed to subscribe /battery_state")
-        (ros::ros-warn "Skip undocking, so please make sure that Fetch is already undocked.")
-        (return))
-      (setq *is-charging* (send msg :is_charging))
-      (if *is-charging*
-        (progn (undock)))
-      (unless *is-charging* (return))
-      (if (eq i 2) (progn (send *ri* :speak "Fail to undock") (ros::ros-error "Fail to undock")))))
+  (let ((undock-success nil))
+    (block go-to-spot-undock
+      (dotimes (i 3)
+        (let* ((msg (one-shot-subscribe "/battery_state" power_msgs::batterystate :timeout 1500))
+               (is-charging (send msg :is_charging)))
```

Comment:
if `msg` is nil, it fails. May be you need to `(setq is-charning ...`) after `(unless msg)`

Reply:
Thank you, I change the code to check if `msg` is `nil` or not